packages/contracts/test/SphinxUtils.t.sol

```diff
@@ -14,6 +15,154 @@ import {
+            chainInfo: VmSafe.ChainInfo(0, 1),
+            kind: VmSafe.AccountAccessKind.Call,
+            account: address(0x11),
+            accessor: address(this),
+            initialized: false,
+            oldBalance: 0,
+            newBalance: 0,
+            deployedCode: new bytes(0),
+            value: 0,
+            data: type(SphinxForkCheck).creationCode,
+            reverted: false,
+            storageAccesses: new Vm.StorageAccess[](0)
+        });
+        accountAccesses[1] = VmSafe.AccountAccess({
+            chainInfo: VmSafe.ChainInfo(0, 1),
+            kind: VmSafe.AccountAccessKind.Create,
+            account: address(expectedAddress),
+            accessor: DETERMINISTIC_DEPLOYMENT_PROXY,
+            initialized: false,
+            oldBalance: 0,
+            newBalance: 0,
+            deployedCode: address(forkCheck).code,
+            value: 0,
+            data: type(SphinxForkCheck).creationCode,
+            reverted: false,
+            storageAccesses: new Vm.StorageAccess[](0)
+        });
+
+        bool passedCheck = checkAccesses(accountAccesses);
+        assertEq(passedCheck, false);
+    }
+
+    function test_sphinxCheckAccesses_fails_incorrect_deployed_code() external {
+        // Deploy the contract so we can get the correct deployed code
+        SphinxForkCheck forkCheck = new SphinxForkCheck();
+        address expectedAddress = vm.computeCreate2Address(
+            0,
+            keccak256(type(SphinxForkCheck).creationCode),
+            DETERMINISTIC_DEPLOYMENT_PROXY
+        );
+
+        Vm.AccountAccess[] memory accountAccesses = new Vm.AccountAccess[](2);
+        accountAccesses[0] = VmSafe.AccountAccess({
+            chainInfo: VmSafe.ChainInfo(0, 1),
+            kind: VmSafe.AccountAccessKind.Call,
+            account: address(DETERMINISTIC_DEPLOYMENT_PROXY),
+            accessor: address(this),
+            initialized: false,
+            oldBalance: 0,
+            newBalance: 0,
+            deployedCode: new bytes(0),
+            value: 0,
+            data: type(SphinxForkCheck).creationCode,
+            reverted: false,
+            storageAccesses: new Vm.StorageAccess[](0)
+        });
+        accountAccesses[1] = VmSafe.AccountAccess({
+            chainInfo: VmSafe.ChainInfo(0, 1),
+            kind: VmSafe.AccountAccessKind.Create,
+            account: address(expectedAddress),
+            accessor: DETERMINISTIC_DEPLOYMENT_PROXY,
+            initialized: false,
+            oldBalance: 0,
+            newBalance: 0,
+            deployedCode: address(forkCheck).code,
+            value: 0,
+            data: type(SphinxForkCheck).creationCode,
+            reverted: false,
+            storageAccesses: new Vm.StorageAccess[](0)
+        });
+
+        bool passedCheck = checkAccesses(accountAccesses);
+        assertEq(passedCheck, false);
+    }
+
+    function test_sphinxCheckAccesses_fails_incorrect_address() external {
```

Comment:
It's not clear what this function is testing. Which variable is different?

Reply:
See inline docs